stub/src/main/java/io/grpc/kotlin/Readiness.kt

```diff
@@ -29,9 +29,12 @@ internal class Readiness(
   private val channel = Channel<Unit>(Channel.CONFLATED)
 
   fun onReady() {
-    if (!channel.offer(Unit)) {
+    try {
+      channel.trySend(Unit).getOrThrow()
+    }
+    catch (e: Exception) {
       throw AssertionError(
-        "Should be impossible; a CONFLATED channel should never return false on offer"
+              "Should be impossible; a CONFLATED channel should never return false on offer"
       )
     }
```

Comment:
Wont this convert whatever exception was previously thrown by `.offer` into an `AssertionError`?

I think the code maintaining previous semantics would be

```kotlin
    channel.trySend(Unit).onFailure { ex ->
      throw ex ?: AssertionError(
        "Should be impossible; a CONFLATED channel should never return false on offer"
      )
    }
```

Reply:
offer shouldn't throw unless the channel is closed, but sure.

Reply:
`.offer` will throw if the channel is closed with an exception eg with `channel.close(VeryBadException())`. I think it will also throw if it is closed normally eg `channel.close()` but I am not sure.

`.offer` will return `false` if the channel is full.

Reply:
In any case `offer` is currently implemented in terms of `trySend`, and throws the exception in the `Failure` if one is present but otherwise returns `false` on `Failure` and `true` on `Success`.